lib/src/layer/polyline_layer.dart

```diff
@@ -39,66 +69,191 @@ class Polyline {
-      isDotted,
-      strokeCap,
-      strokeJoin,
-      useStrokeWidthInMeter);
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      (other is Polyline<TapKeyType> &&
+          listEquals(points, other.points) &&
+          strokeWidth == other.strokeWidth &&
+          color == other.color &&
+          borderStrokeWidth == other.borderStrokeWidth &&
+          borderColor == other.borderColor &&
+          listEquals(gradientColors, other.gradientColors) &&
+          listEquals(colorsStop, other.colorsStop) &&
+          isDotted == other.isDotted &&
+          strokeCap == other.strokeCap &&
+          strokeJoin == other.strokeJoin &&
+          useStrokeWidthInMeter == other.useStrokeWidthInMeter &&
+          hitKey == other.hitKey);
+
+  List<Object?> get _baseHashableValues => [
+        strokeWidth,
+        color,
+        borderStrokeWidth,
+        borderColor,
+        gradientColors,
+        colorsStop,
+        isDotted,
+        strokeCap,
+        strokeJoin,
+        useStrokeWidthInMeter,
+        hitKey,
+      ];
+
+  /// Used to batch draw calls to the canvas
+  int get renderHashCode => Object.hashAll(_baseHashableValues);
+
+  @override
+  int get hashCode => Object.hashAll([points, ..._baseHashableValues]);
 }
 
 @immutable
-class PolylineLayer extends StatelessWidget {
-  final List<Polyline> polylines;
-  final bool polylineCulling;
+class PolylineLayer<TapKeyType extends Object> extends StatelessWidget {
+  final List<Polyline<TapKeyType>> polylines;
+
+  /// A notifier to notify when a hit is detected over a/multiple [Polyline]s
+  ///
+  /// To listen for hits, wrap the layer in a standard hit detector widget, such
+  /// as [GestureDetector] and/or [MouseRegion] (and set
+  /// [HitTestBehavior.deferToChild] if necessary). Then use the latest value
+  /// (via [ValueNotifier.value]) in the detector's callbacks. It is also
+  /// possible to listen to the notifier directly.
+  ///
+  /// A [Polyline.hitKey] may be used to attach additional retrievable
+  /// information, or to signify whether a [Polyline] is "tappable", for example.
+  ///
+  /// Note that a hover event is included as a hit event. Therefore for
+  /// performance reasons, it may be advantageous to check the new value's
+  /// equality against the previous value (excluding the [PolylineHit.point],
+  /// which will always change), and avoid doing heavy work if they are the same.
+  ///
+  /// See online documentation for more detailed usage instructions. See the
+  /// example project for an example implementation.
+  ///
+  /// Will notify [PolylineHit]s if any [Polyline]s were hit, otherwise will
+  /// notify `null`.
+  final PolylineHitNotifier<TapKeyType>? hitNotifier;
+
+  /// The number of pixels away from a visual line (including any width and
+  /// outline) in which a hit should still register as a hit on the line, and
+  /// trigger a notification to [hitNotifier]
+  ///
+  /// Defaults to 3.
```

Comment:
It seems quite useful to have this configurable but I would suggest to make the default 0. Larger values can make for a nicer UX when you have non-overlapping lines but they would probably be quite annoying for overlapping lines especially if they're parallel.

Reply:
But you have to balance with accessibility. I think 3px is alright.

Reply:
Thinking about this a bit more maybe this should be posed as "min dimension". There's no point in adding a transparent out-out-line to a line that might transparently occlude a neighbor, if the line is already wide.

Reply:
Actually yeah, a minimum would make some sense. I'll have a play around and see what's comfortable to me, then add a little bit more.

Reply:
Resolved.